File: Typescript/Ejercicio 2/GestionPlatos.ts
import { Plato } from "./Plato"

export class GestionPlatos {
    private menu: Map<string, Plato> = new Map();

    agregarPlato(nombre: string, precio: number): void {
        if (this.menu.has(nombre)) {
            console.log("El plato ya está en el menú");
        } else {
            this.menu.set(nombre, new Plato(nombre, precio));
            console.log("Plato agregado exitosamente.");
        }
    }

    mostrarMenu(): void {
        console.log("\n--- Menú de Platos ---");
        if (this.menu.size === 0) {
            console.log("No hay platos registrados.");
        } else {
            this.menu.forEach(plato => console.log(plato.toString()));
        }
    }

    existePlato(nombre: string): boolean {
        return this.menu.has(nombre);
    }

    obtenerPrecio(nombre: string): number | undefined {
        return this.menu.get(nombre)?.precio;
    }
}

File: Go/Ejercicio 2/gestion_platos.go
package main

import "fmt"

type GestionPlatos struct {
	Menu map[string]Plato
}

func NewGestionPlatos() *GestionPlatos {
	return &GestionPlatos{Menu: make(map[string]Plato)}
}

func (gp *GestionPlatos) AgregarPlato(nombre string, precio float64) {
	if _, existe := gp.Menu[nombre]; existe {
		fmt.Println("El plato ya existe en el menú.")
	} else {
		gp.Menu[nombre] = Plato{Nombre: nombre, Precio: precio}
		fmt.Println("Plato agregado exitosamente.")
	}
}

func (gp *GestionPlatos) MostrarMenu() {
	fmt.Println("\n--- Menú de Platos ---")
	if len(gp.Menu) == 0 {
		fmt.Println("No hay platos registrados.")
	} else {
		for _, plato := range gp.Menu {
			fmt.Println(plato)
		}
	}
}

func (gp *GestionPlatos) ExistePlato(nombre string) bool {
	_, existe := gp.Menu[nombre]
	return existe
}

func (gp *GestionPlatos) ObtenerPrecio(nombre string) float64 {
	if plato, existe := gp.Menu[nombre]; existe {
		return plato.Precio
	}
	return 0
}
